raiden/blockchain_events_handler.py

```diff
@@ -326,17 +326,20 @@ def handle_channel_closed(raiden: "RaidenService", event: Event):
             raiden.handle_and_track_state_change(channel_closed)
         else:
             # Must be a light client
+            closing_participant = args["closing_participant"]
+            non_closing_participant = channel_state.partner_state.address \
+                if channel_state.our_state.address == closing_participant \
+                else channel_state.our_state.address
             latest_non_closing_balance_proof = LightClientMessageHandler.get_latest_light_client_non_closing_balance_proof(
-                channel_state.identifier, channel_state.partner_state.address, raiden.wal.storage
+                channel_state.identifier, non_closing_participant, raiden.wal.storage
```

Comment:
previously this was channel_state.our_state.address

then we used channel_state.partner_state.address 

did you find wich scenarios (toppologies) were causing an issue when using our state?

i mean, i found it ok now, but we should have clearly identified the wrong behaviour when making a critical change like this

what do you guys think? @mortelli @JONAF2103

Reply:
channel_state.our_state.address worked fine with lc-lc same hub, but failed in lc-fn-lc.

channel_state.partner_state.address worked well in any situation, but just because some checks that were done in the ContractReceiveChannelClosedLight handler, which prevented the wrong participant from sending the non closis bp.

I changed it to the current code in order to make it cleaner, less confussing, and much less error prone in the future.

Reply:
My question is why handle_channel_closed is being triggered by both participants, and if that's the correct behaviour then what should be latest_non_closing_balance_proof? if that is the balance proof for the non closing participant then the new logic doesn't make any sense to me, since you always will need the address for the non closing participant, if not then i miss understood the purpose of this logic.

Reply:
I guess the closing participant needs to get notified that the transaction it sent was successful, in order to change update the channel state.

Reply:
yeah basically both sides must update the channel status

but the hub must send the latest non closing balance proof for the non closing participant (acting as a watchtower) in order to prevent the closer to get all the funds